### PR TITLE
[WIP] Flutter Web (WIP, don't merge)

### PR DESCRIPTION
**Pull request by rostopira**
Adds support for Flutter Web

- [x] P2P
- [x] Data Channel (text)
- [x] Data Channel (binary)
- [x] getUserMedia
- [x] getDisplayMedia
- [x] getSources
- [x] Sending video/audio
- [x] Displaying video/audio (waiting for pr merge)
- [x] Capture frame (works only once for no reason)
- [x] Video/audio recording (gives blob)
- [ ] Camera switch

Linked issue #85

### CODE REVIEW

**Comment by rostopira** [comment]
@cloudwebrtc As you can see, there is some code duplication
What about moving all enums to another file like `enums.dart`?

**Comment by rostopira** [comment]
Note: in every file you can see imports:
```dart
// ignore: uri_does_not_exist
import 'dart:html' as HTML;
// ignore: uri_does_not_exist
import 'dart:js' as JS;
```
This is required to force Dart Analyzer to shut up

**Comment by cloudwebrtc** [comment]
> @cloudwebrtc As you can see, there is some code duplication
> What about moving all enums to another file like `enums.dart`?

Yes, this is a good idea.

**Comment by rostopira** [comment]
`HtmlElementView` is now available on master, but...

**Comment by cloudwebrtc** [comment]
Wow, cool, I will try it right away.

**Comment by rostopira** [comment]
Don't worth it right now, wait a few days, I think everything else that required will be in master by EOW. For now it's working only with ugly hacks, as I said, this branch won't work

**Comment by rostopira** [comment]
@cloudwebrtc  You can try it now. But it's copy-paste from my local fork, probably needs some fixes, but I don't have much time right now, gonna fix it on weekend

File: lib/web/rtc_video_view.dart [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  static void fixVideoElements() =>
+    _videoViews.values.forEach((v) => v.play());
+
+  /// Currently contains hacky solution
+  /// Multiple videos won't work
+  /// Waiting for onPlatformViewCreated callback implementation
+  initialize() async {
+    if (!_isViewFactoryRegistered) {
+      // ignore: implementation_imports
+      ui.platformViewRegistry.registerViewFactory('webrtc_video', (int viewId) {
```
Comment (rostopira) [inline lib/web/rtc_video_view.dart]: Dart Analyzer will yell on this, just ignore it

**Comment by cloudwebrtc** [comment]
@rostopira Great, just now, I tried the branch and getUserMedia works fine.

**Comment by rostopira** [comment]
@cloudwebrtc For now I can't get multiple videos, video needs additional `play()` if you navigated to another page and back, multiple issues with frame capture and recording on OS X (windows is fine), everything else is pretty working in my project (P2P/video/audio)

**Comment by cloudwebrtc** [comment]
I tried to debug it, but vscode doesn't seem to be able to debug the flutter web project, I will try again on the weekend.

**Comment by cloudwebrtc** [comment]
@rostopira I have modified getDisplayMedia to get the screen object, but I get it to JsObject. I haven't found a way to convert this JsObject to MediaStream. Because it is stored in a private variable.

**Comment by cloudwebrtc** [comment]
`getDisplayMedia` is working fine.

**Comment by cloudwebrtc** [comment]
Switch camera in Chrome will trigger SDP re-negotiation, I have not found a better way. :(

**Pull request merged**